### PR TITLE
feat: Aave + Spark Protocol Plugins for refinance

### PR DESCRIPTION
**Pull request by robercano**
- Added Aave + Spark plugins, actions wrapper and action builders
- Added unit tests for the new action wrappers and builders
- Updated refinance strategies to use the new `OpenPosition` step

### CODE REVIEW

**Review by robercano — COMMENTED** [review]
@paszkowskiDamian looks good to me! I cannot approve, but I didn't see anything that needed changing

File: sdk/simulator-service/src/implementation/simulator-engine/reducer/depositBorrowReducer.ts [hunk 1]
```diff
@@ -1,31 +1,28 @@
...
+import { addBalance, subtractBalance, getValueFromReference } from '../../utils'
 import { ISimulationState } from '../../../interfaces/simulation'
 
 export function depositBorrowReducer(
   step: steps.DepositBorrowStep,
   state: ISimulationState,
 ): ISimulationState {
   const afterDeposit = subtractBalance(
-    getReferencedValue(step.inputs.depositAmount),
+    getValueFromReference(step.inputs.depositAmount),
```
Comment (piotrwitek) [inline sdk/simulator-service/src/implementation/simulator-engine/reducer/depositBorrowReducer.ts:11]: just a small readability improvement would be to assign `getValueFromReference(step.inputs.depositAmount)` to a variable as it is referenced 2 times, same for `getValueFromReference(step.inputs.borrowAmount)`

File: sdk/simulator-service/src/strategies/refinanceAnyPair/RefinanceLendingToLendingAnyPair.ts [hunk 2]
```diff
@@ -65,11 +53,14 @@ export async function refinanceLendingToLendingAnyPair(
       type: SimulationSteps.Flashloan,
       inputs: {
         amount: flashloanAmount,
-        provider: FlashloanProvider.Maker,
+        provider:
+          flashloanAmount.token.symbol === 'DAI'
```
Comment (piotrwitek) [inline sdk/simulator-service/src/strategies/refinanceAnyPair/RefinanceLendingToLendingAnyPair.ts:57]: would be better to have reusable constant for it

File: sdk/simulator-service/src/strategies/refinanceAnyPair/Strategy.ts [hunk 3]
```diff
@@ -3,35 +3,48 @@ import { makeStrategy } from '../../implementation/utils'
...
     step: SimulationSteps.Swap,
     optional: true,
   },
   {
+    name: 'OpenTargetPosition',
+    step: SimulationSteps.OpenPosition,
+    optional: false,
+  },
+  {
+    name: 'DepositBorrowToTargetPosition',
```
Comment (piotrwitek) [inline sdk/simulator-service/src/strategies/refinanceAnyPair/Strategy.ts:26]: great change here, basically code is self documenting now :)

File: sdk/simulator-service/src/strategies/refinanceAnyPair/RefinanceLendingToLendingAnyPair.ts [hunk 4]
```diff
@@ -95,13 +87,20 @@ export async function refinanceLendingToLendingAnyPair(
...
+    }))
     .next(async (ctx) => ({
-      name: 'DepositBorrowToTarget',
+      name: 'DepositBorrowToTargetPosition',
       type: SimulationSteps.DepositBorrow,
       inputs: {
         // refactor
         borrowAmount: isDebtSwapSkipped
-          ? ctx.getReference(['PaybackWithdrawFromSource', 'paybackAmount'])
+          ? ctx.getReference(['SwapCollateralFromSourcePosition', 'received'])
```
Comment (piotrwitek) [inline sdk/simulator-service/src/strategies/refinanceAnyPair/RefinanceLendingToLendingAnyPair.ts:103]: small naming consistenty improvements are really nice adding clarity and additional context

**Review by piotrwitek — APPROVED** [review]
Stellar work, huge amount of type and consistency improvements across the board, many improvements to the code to be self documenting by adding context with a more verbose naming conventions, which is great. Added few really bad nit-picks, but there was nothing else to comment on except the high readability and consistency.

File: sdk/simulator-service/src/strategies/refinanceAnyPair/RefinanceLendingToLendingAnyPair.ts (hunk 2 above)
Comment (robercano) [reply sdk/simulator-service/src/strategies/refinanceAnyPair/RefinanceLendingToLendingAnyPair.ts:57]: We do have `CommonTokenSymbols`. I'll refactor it

File: sdk/simulator-service/src/strategies/refinanceAnyPair/RefinanceLendingToLendingAnyPair.ts (hunk 4 above)
Comment (robercano) [reply sdk/simulator-service/src/strategies/refinanceAnyPair/RefinanceLendingToLendingAnyPair.ts:103]: 👏

File: sdk/simulator-service/src/types/Types.ts [hunk 5]
```diff
@@ -6,5 +6,5 @@ export type StepName = string
 
 export type BalancesRecord = Record<AddressValue, ITokenAmount>
 export type PositionsRecord = Record<string, IPosition>
-export type StepsRecord = Record<StepName, steps.Steps>
-export type SwapsRecord = Record<string, SimulatedSwapData>
+export type StepsArray = steps.Steps[]
+export type SwapArray = SimulatedSwapData[]
```
[the remark this answers is not in the document]
Comment (robercano) [reply sdk/simulator-service/src/types/Types.ts:10]: This is Damian's change, but I agree we should be consistent there. I'll change it to plural

**Review by piotrwitek — APPROVED** [review]

**Pull request merged**